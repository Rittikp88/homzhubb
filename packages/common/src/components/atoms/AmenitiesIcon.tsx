import React from 'react';
import { StyleProp, StyleSheet, View, ViewStyle } from 'react-native';
import Icon from '@homzhub/common/src/assets/icon';
import { theme } from '@homzhub/common/src/styles/theme';
import { Text } from '@homzhub/common/src/components/atoms/Text';
import { Divider } from '@homzhub/common/src/components/atoms/Divider';

interface IProps {
  direction: 'row' | 'column';
  icon: string;
  iconSize?: number;
  iconColor?: string;
  label: string;
  isLastIndex?: boolean;
  containerStyle?: StyleProp<ViewStyle>;
}

const AmenitiesIcon = (props: IProps): React.ReactElement => {
  const {
    direction,
    icon,
    iconSize = 40,
    iconColor = theme.colors.darkTint3,
    label,
    isLastIndex = false,
    containerStyle,
  } = props;

  const renderText = (): React.ReactElement => {
    const labelStyle = direction === 'column' ? styles.columnLabel : styles.rowLabel;
    return (
      <Text type="small" textType="regular" style={[styles.label, labelStyle]}>
        {label}
      </Text>
    );
  };

  const renderColumn = (): React.ReactElement => {
    return (
      <>
        <View style={styles.columnContainer}>
          <Icon name={icon} size={iconSize} color={iconColor} />
          {renderText()}
        </View>
        {!isLastIndex && direction === 'column' && <Divider containerStyles={styles.divider} />}
      </>
    );
  };

  const renderRow = (): React.ReactElement => {
    return (
      <View style={[styles.rowContainer, containerStyle]}>
        <Icon name={icon} size={iconSize} color={iconColor} />
        {renderText()}
      </View>
    );
  };

  return direction === 'column' ? renderColumn() : renderRow();
};

export { AmenitiesIcon };

const styles = StyleSheet.create({
  columnContainer: {
    flexDirection: 'column',
    justifyContent: 'space-around',
    alignItems: 'center',
    marginHorizontal: 10,
  },
  rowContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  label: {
    color: theme.colors.darkTint3,
  },
  columnLabel: {
    alignItems: 'center',
  },
  rowLabel: {
    marginStart: 6,
  },
  divider: {
    borderWidth: 1,
    borderColor: theme.colors.background,
    height: 25,
  },
});
