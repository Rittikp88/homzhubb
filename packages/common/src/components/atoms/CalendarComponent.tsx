import React, { Component } from 'react';
import { Calendar, DateObject } from 'react-native-calendars';
import { View, FlatList, TouchableOpacity, StyleSheet, StyleProp, TextStyle, ViewStyle } from 'react-native';
import moment from 'moment';
import { DateFormats, DateUtils, MonthNames } from '@homzhub/common/src/utils/DateUtils';
import { PlatformUtils } from '@homzhub/common/src/utils/PlatformUtils';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { theme } from '@homzhub/common/src/styles/theme';
import { Divider, Button, Text, Label, WithShadowView } from '@homzhub/common/src/components';

interface ICalendarProps {
  onSelect: (day: string) => void;
  selectedDate: string;
}

interface ICalendarState {
  isMonthView: boolean;
  selectedDate: string;
  day: string;
  month: number;
  year: string;
}

export class CalendarComponent extends Component<ICalendarProps, ICalendarState> {
  public constructor(props: ICalendarProps) {
    super(props);
    const { selectedDate } = props;
    this.state = {
      isMonthView: false,
      selectedDate,
      day: moment().format(DateFormats.DD),
      month: new Date().getMonth(),
      year: moment().format(DateFormats.YYYY),
    };
  }

  public render(): React.ReactNode {
    const { isMonthView } = this.state;
    return (
      <View>
        {this.renderHeader()}
        {isMonthView ? this.renderMonthView() : this.renderCalendar()}
      </View>
    );
  }

  private renderHeader = (): React.ReactElement => {
    const { year, month, isMonthView, selectedDate } = this.state;
    const newMonth = moment(selectedDate).month();
    const newYear = moment(selectedDate).year();
    const updateMonth = DateUtils.getFullMonthName(newMonth || month, DateFormats.MMMM);

    return (
      <>
        <View style={styles.headerContainer}>
          <Icon
            name={icons.leftArrow}
            onPress={this.handleBackPress}
            size={22}
            color={month === moment().month() ? theme.colors.disabled : theme.colors.primaryColor}
          />
          <Text
            type="small"
            textType="semiBold"
            onPress={this.handleMonthPress}
            style={StyleSheet.flatten([customStyles.headerTitle(isMonthView)])}
          >
            {`${updateMonth} ${newYear || year}`}
          </Text>
          <Icon name={icons.rightArrow} size={22} color={theme.colors.primaryColor} onPress={this.handleNextPress} />
        </View>
        <Divider />
      </>
    );
  };

  private renderMonthView = (): React.ReactNode => {
    return (
      <FlatList
        data={MonthNames}
        renderItem={this.renderItem}
        contentContainerStyle={styles.listContent}
        keyExtractor={this.renderKeyExtractor}
        numColumns={4}
      />
    );
  };

  private renderKeyExtractor = (item: string, index: number): string => {
    return `${item}-${index}`;
  };

  private renderItem = ({ item, index }: { item: string; index: number }): React.ReactElement => {
    const onPressItem = (): void => this.onSelectMonth(item, index);
    const { month } = this.state;
    const isSelected = month === index;
    return (
      <TouchableOpacity
        key={index}
        style={StyleSheet.flatten([customStyles.renderItemView(isSelected)])}
        onPress={onPressItem}
      >
        <Label type="large" style={StyleSheet.flatten([customStyles.renderItemTitle(isSelected)])}>
          {item}
        </Label>
      </TouchableOpacity>
    );
  };

  private renderCalendar = (): React.ReactElement => {
    const { day, month, year, selectedDate } = this.state;
    const updateMonth = DateUtils.getFullMonthName(month, DateFormats.MMM);
    const date =
      selectedDate || DateUtils.getFormattedDate(day, updateMonth, year, DateFormats.YYYYMMMDD).toDateString();
    return (
      <>
        <Calendar
          hideArrows
          // @ts-ignore
          renderHeader={this.headerView}
          minDate={new Date()}
          current={date}
          key={date}
          style={styles.calendarStyle}
          onDayPress={this.onDayPress}
          markingType="custom"
          theme={{}}
          markedDates={{
            [date]: {
              customStyles: {
                container: {
                  backgroundColor: theme.colors.primaryColor,
                  borderRadius: 4,
                },
                text: {
                  color: theme.colors.white,
                },
              },
            },
          }}
        />
        <WithShadowView outerViewStyle={styles.shadowView}>
          <Button type="primary" title="Select" containerStyle={styles.buttonStyle} onPress={this.handleSelect} />
        </WithShadowView>
      </>
    );
  };

  private onSelectMonth = (item: string, index: number): void => {
    const { year } = this.state;
    this.setState({ month: index, isMonthView: false });
    this.getSelectedDate(index, Number(year));
  };

  private onDayPress = (day: DateObject): void => {
    this.setState({ selectedDate: day.dateString });
  };

  private handleSelect = (): void => {
    const { onSelect } = this.props;
    const { selectedDate } = this.state;
    onSelect(selectedDate);
  };

  private headerView = (): null => {
    return null;
  };

  private handleBackPress = (): void => {
    const { year, isMonthView, month } = this.state;

    if (month === moment().month()) {
      return;
    }

    if (isMonthView) {
      const previousYear = Number(year) - 1;
      this.getSelectedDate(month, previousYear);
      this.setState({ year: previousYear.toString() });
    } else {
      const previousMonth = month - 1;
      this.getSelectedDate(previousMonth, Number(year));
      this.setState({ month: previousMonth });
    }
  };

  private handleNextPress = (): void => {
    const { year, isMonthView, month } = this.state;

    if (isMonthView) {
      const nextYear = Number(year) + 1;
      this.getSelectedDate(month, nextYear);
      this.setState({ year: nextYear.toString() });
    } else {
      const nextMonth = month + 1;
      this.getSelectedDate(nextMonth, Number(year));
      this.setState({ month: nextMonth });
    }
  };

  private getSelectedDate = (month: number, year: number): void => {
    const { selectedDate } = this.props;
    const isNotSelected = moment(selectedDate).month() !== month || moment(selectedDate).year() !== year;
    if (isNotSelected) {
      this.setState({ selectedDate: '' });
    } else {
      this.setState({ selectedDate });
    }
  };

  private handleMonthPress = (): void => {
    const { isMonthView } = this.state;
    this.setState({ isMonthView: !isMonthView });
  };
}

const styles = StyleSheet.create({
  shadowView: {
    paddingTop: 10,
    marginBottom: PlatformUtils.isIOS() ? 20 : 0,
    paddingBottom: 0,
  },
  buttonStyle: {
    flex: 0,
    margin: 16,
  },
  headerContainer: {
    justifyContent: 'space-between',
    flexDirection: 'row',
    paddingVertical: 20,
    paddingHorizontal: 24,
  },
  listContent: {
    marginVertical: 16,
  },
  calendarStyle: {
    height: 310,
  },
});

const customStyles = {
  headerTitle: (isMonthView: boolean): StyleProp<TextStyle> => ({
    color: isMonthView ? theme.colors.darkTint2 : theme.colors.primaryColor,
  }),
  renderItemView: (isSelected: boolean): StyleProp<ViewStyle> => ({
    width: 80,
    marginVertical: 12,
    marginHorizontal: 10,
    alignItems: 'center',
    backgroundColor: isSelected ? theme.colors.primaryColor : theme.colors.white,
    borderRadius: 4,
  }),
  renderItemTitle: (isSelected: boolean): StyleProp<TextStyle> => ({
    paddingVertical: 6,
    color: isSelected ? theme.colors.white : theme.colors.darkTint2,
  }),
};
