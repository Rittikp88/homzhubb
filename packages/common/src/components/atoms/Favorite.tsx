import React from 'react';
import { ViewStyle, StyleProp, TouchableOpacity } from 'react-native';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { theme } from '@homzhub/common/src/styles/theme';

interface IProps {
  onFavorite: () => void;
  isFavorite: boolean;
  iconColor?: string;
  iconSize?: number;
  containerStyle?: StyleProp<ViewStyle>;
}

const Favorite = (props: IProps): React.ReactElement => {
  const { onFavorite, isFavorite, iconColor = theme.colors.darkTint6, containerStyle, iconSize = 32 } = props;

  const onFavoritePress = (): void => {
    onFavorite();
  };

  return (
    <TouchableOpacity style={containerStyle} onPress={onFavoritePress}>
      <Icon
        name={isFavorite ? icons.filledHeart : icons.heartOutline}
        size={iconSize}
        color={isFavorite ? theme.colors.favourite : iconColor}
        testID="icon"
      />
    </TouchableOpacity>
  );
};

export { Favorite };
