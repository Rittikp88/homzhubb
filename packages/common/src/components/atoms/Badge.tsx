import React from 'react';
import { StyleProp, StyleSheet, View, ViewStyle } from 'react-native';
import { theme } from '@homzhub/common/src/styles/theme';
import { FontWeightType, Label } from '@homzhub/common/src/components/atoms/Text';

interface IProps {
  title: string;
  badgeColor: string;
  badgeStyle?: StyleProp<ViewStyle>;
  textType?: FontWeightType;
}

export const Badge = (props: IProps): React.ReactElement => {
  const { badgeColor, badgeStyle, title, textType } = props;
  return (
    <View style={[styles.badge, badgeStyle, { backgroundColor: badgeColor }]}>
      <Label type="regular" textType={textType} style={styles.title}>
        {title}
      </Label>
    </View>
  );
};

const styles = StyleSheet.create({
  badge: {
    borderRadius: 2,
    paddingHorizontal: 6,
  },
  title: {
    color: theme.colors.white,
    alignSelf: 'center',
  },
});
