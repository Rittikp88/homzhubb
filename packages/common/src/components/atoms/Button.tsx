import React from 'react';
import {
  GestureResponderEvent,
  ImageStyle,
  StyleProp,
  StyleSheet,
  TextStyle,
  TouchableOpacity,
  ViewStyle,
} from 'react-native';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon from '@homzhub/common/src/assets/icon';
import { FontWeightType, Label, Text, TextFieldType, TextSizeType } from '@homzhub/common/src/components/atoms/Text';

export type ButtonType = 'primary' | 'secondary' | 'secondaryOutline';

export interface IButtonProps {
  type: ButtonType;
  node?: React.ReactNode;
  children?: React.ReactNode;
  title?: string;
  onPress?: (event?: GestureResponderEvent) => void;
  disabled?: boolean;
  activeOpacity?: number;
  titleStyle?: StyleProp<TextStyle>;
  containerStyle?: StyleProp<ViewStyle>;
  textType?: TextFieldType;
  textSize?: TextSizeType;
  fontType?: FontWeightType;
  icon?: string;
  iconSize?: number;
  iconColor?: string;
  iconStyle?: StyleProp<ImageStyle>;
  testID?: string;
}

export class Button extends React.PureComponent<IButtonProps> {
  public render = (): React.ReactElement => {
    const {
      onPress,
      disabled = false,
      activeOpacity = 0.5,
      title,
      icon,
      iconSize,
      iconColor,
      iconStyle,
      testID,
      node,
      children,
    } = this.props;
    return (
      <TouchableOpacity
        activeOpacity={activeOpacity}
        onPress={onPress}
        disabled={disabled}
        style={this.getContainerStyle()}
        testID={testID}
      >
        {title && this.getTextField()}
        {!!icon && <Icon name={icon} size={iconSize} color={iconColor} style={iconStyle} />}
        {node && node}
        {children && children}
      </TouchableOpacity>
    );
  };

  private getTextField = (): React.ReactElement => {
    const { textType, title, textSize = 'small', fontType = 'semiBold' } = this.props;
    let TextField = Text;

    if (textType === 'label') {
      TextField = Label;
    }

    return (
      <TextField type={textSize} textType={fontType} style={this.getTextStyle()}>
        {title}
      </TextField>
    );
  };

  private getContainerStyle = (): ViewStyle => {
    const { disabled, type, containerStyle = {} } = this.props;
    let themedStyle: StyleProp<ViewStyle> = styles.primary;
    if (disabled) {
      themedStyle = styles.disabled;
    }

    if (type === 'secondary' && !disabled) {
      themedStyle = styles.secondary;
    }

    if (type === 'secondaryOutline' && !disabled) {
      themedStyle = styles.secondaryOutline;
    }
    return StyleSheet.flatten([styles.container, themedStyle, containerStyle]);
  };

  private getTextStyle = (): TextStyle => {
    const { type, disabled = false, titleStyle = {} } = this.props;
    let themedStyle = {};
    if (type === 'secondary' && !disabled) {
      themedStyle = { color: theme.colors.active };
    }
    return StyleSheet.flatten([styles.textStyle, themedStyle, titleStyle]);
  };
}

const styles = StyleSheet.create({
  container: {
    ...theme.globalStyles.center,
    borderRadius: 4,
  },
  textStyle: {
    textAlign: 'center',
    color: theme.colors.white,
    marginVertical: 12,
    marginHorizontal: 36,
  },
  disabled: {
    backgroundColor: theme.colors.disabled,
  },
  primary: {
    backgroundColor: theme.colors.primaryColor,
  },
  secondary: {
    borderWidth: 1,
    borderColor: theme.colors.active,
    backgroundColor: theme.colors.white,
  },
  secondaryOutline: {
    borderWidth: 0.5,
    borderColor: theme.colors.white,
  },
});
