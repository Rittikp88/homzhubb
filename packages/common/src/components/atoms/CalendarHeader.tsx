import React from 'react';
import { StyleSheet, View } from 'react-native';
import moment from 'moment';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { theme } from '@homzhub/common/src/styles/theme';
import { Divider } from '@homzhub/common/src/components/atoms/Divider';
import { Text } from '@homzhub/common/src/components/atoms/Text';

interface IProps {
  isAllowPastDate?: boolean;
  headerTitle?: string;
  headerYear?: string;
  yearTitle?: string;
  isMonthView?: boolean;
  isYearView?: boolean;
  isCurrentMonth?: boolean;
  maxDate?: string;
  month?: number;
  onBackPress?: () => void;
  onNextPress?: () => void;
  onMonthPress?: () => void;
  onYearPress?: () => void;
}

const CalendarHeader = (props: IProps): React.ReactElement => {
  const {
    isAllowPastDate,
    maxDate,
    onBackPress,
    onNextPress,
    onMonthPress,
    onYearPress,
    isCurrentMonth,
    headerTitle,
    headerYear,
    isMonthView = false,
    isYearView = false,
    yearTitle,
    month,
  } = props;
  const isNextDisable =
    maxDate && (moment(maxDate).month() === moment().month() || (month && moment(maxDate).month() === month));
  return (
    <>
      <View style={styles.headerContainer}>
        <Icon
          name={icons.leftArrow}
          onPress={onBackPress}
          size={22}
          color={!isAllowPastDate && isCurrentMonth && !isYearView ? theme.colors.disabled : theme.colors.primaryColor}
        />
        {!isYearView && !isMonthView && (
          <Text type="small" textType="semiBold" onPress={onMonthPress} style={styles.headerTitle}>
            {headerTitle}
          </Text>
        )}
        {isMonthView && (
          <Text type="small" textType="semiBold" onPress={onYearPress} style={styles.headerTitle}>
            {headerYear}
          </Text>
        )}
        {isYearView && (
          <Text type="small" textType="semiBold" style={styles.headerTitle}>
            {yearTitle}
          </Text>
        )}
        <Icon
          name={icons.rightArrow}
          size={22}
          color={isNextDisable ? theme.colors.disabled : theme.colors.primaryColor}
          onPress={onNextPress}
        />
      </View>
      <Divider />
    </>
  );
};

export default CalendarHeader;

const styles = StyleSheet.create({
  headerContainer: {
    justifyContent: 'space-between',
    flexDirection: 'row',
    paddingVertical: 20,
    paddingHorizontal: 24,
  },
  headerTitle: {
    color: theme.colors.primaryColor,
  },
});
