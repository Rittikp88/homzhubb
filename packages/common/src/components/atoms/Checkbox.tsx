import React from 'react';
import { StyleProp, StyleSheet, TextStyle, TouchableOpacity, View, ViewStyle } from 'react-native';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { Label, TextSizeType } from '@homzhub/common/src/components/atoms/Text';

interface ICheckboxOptions {
  selected: boolean;
  label: string;
  onToggle: () => void;
  containerStyle?: StyleProp<ViewStyle>;
  labelStyle?: StyleProp<TextStyle>;
  iconSize?: number;
  iconStyle?: object;
  iconSelectedStyle?: object;
  labelType?: TextSizeType;
}

const RNCheckbox = (props: ICheckboxOptions): React.ReactElement => {
  const {
    label = '',
    selected = false,
    containerStyle = {},
    labelStyle = {},
    iconSize = 22,
    iconStyle = {},
    iconSelectedStyle = {},
    labelType = 'large',
    onToggle,
  } = props;
  return (
    <View style={[styles.container, containerStyle]}>
      <TouchableOpacity onPress={onToggle}>
        <Icon
          name={selected ? icons.checkboxOn : icons.checkboxOff}
          size={iconSize}
          color={selected ? theme.colors.primaryColor : theme.colors.disabled}
          style={selected ? iconSelectedStyle : iconStyle}
        />
      </TouchableOpacity>
      <Label type={labelType} textType="regular" style={[styles.label, labelStyle]}>
        {label}
      </Label>
    </View>
  );
};

export { RNCheckbox };

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  label: {
    marginStart: 12,
  },
});
