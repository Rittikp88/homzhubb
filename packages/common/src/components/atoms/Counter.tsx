import React, { ReactElement, useEffect, useState } from 'react';
import { StyleProp, StyleSheet, View, ViewStyle } from 'react-native';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { SVGUri } from '@homzhub/common/src/components/atoms/Svg';
import { Label, Text } from '@homzhub/common/src/components/atoms/Text';

interface ITitle {
  title: string;
  id?: number;
}

export interface ICounterProps {
  defaultValue: number;
  name?: ITitle;
  svgImage?: string;
  onValueChange: (count: number, id?: number) => void;
  maxCount?: number;
  minCount?: number;
  containerStyles?: StyleProp<ViewStyle>;
  testID?: string;
}

export const Counter = (props: ICounterProps): React.ReactElement => {
  const { onValueChange, defaultValue, name, svgImage, maxCount = 10, minCount = 0, containerStyles } = props;
  const [count, setCount] = useState(defaultValue);

  useEffect(() => {
    onValueChange(count, name?.id);
  }, [count]);

  const incrementCount = (): void => {
    if (count < maxCount) {
      setCount((prev) => prev + 1);
    }
  };

  const decrementCount = (): void => {
    if (count > minCount) {
      setCount((prev) => prev - 1);
    }
  };

  const renderCounter = (): ReactElement => {
    return (
      <View style={[styles.counterContainer, styles.rowStyle]}>
        <Icon name={icons.minus} color={theme.colors.primaryColor} size={20} onPress={decrementCount} />
        <Label type="large">{count}</Label>
        <Icon name={icons.plus} color={theme.colors.primaryColor} size={20} onPress={incrementCount} />
      </View>
    );
  };

  return (
    <View style={[styles.rowStyle, containerStyles]}>
      <View style={styles.imageContainer}>
        {svgImage && <SVGUri height={20} width={20} uri={svgImage} />}
        {name && (
          <Text style={styles.textStyle} type="small">
            {name.title}
          </Text>
        )}
      </View>
      {renderCounter()}
    </View>
  );
};

const styles = StyleSheet.create({
  rowStyle: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  counterContainer: {
    borderColor: theme.colors.darkTint6,
    borderWidth: 1,
    borderRadius: 4,
    width: 100,
    padding: 4,
  },
  imageContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  textStyle: {
    marginLeft: 12,
  },
});
