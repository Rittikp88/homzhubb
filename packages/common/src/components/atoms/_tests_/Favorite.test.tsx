import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Favorite } from '@homzhub/common/src/components/atoms/Favorite';

const mock = jest.fn();

describe('Favorite', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    const props = {
      onFavorite: mock,
      isFavorite: true,
    };
    wrapper = shallow(<Favorite {...props} />);
  });

  it('should match snapshot for selected true', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
