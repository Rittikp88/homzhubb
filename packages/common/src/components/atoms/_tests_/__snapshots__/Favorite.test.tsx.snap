// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite should match snapshot for selected true 1`] = `
<ForwardRef
  onPress={[Function]}
>
  <Icon
    allowFontScaling={false}
    color="#EA464C"
    name="filled-heart"
    size={32}
    testID="icon"
  />
</ForwardRef>
`;
